packages/sync/src/modules/class-full-sync-immediately.php

```diff
@@ -0,0 +1,405 @@
+		 * and sent to the server so that it knows a full sync is coming.
+		 *
+		 * @param array $full_sync_config Sync configuration for all sync modules.
+		 * @param array $range Range of the sync items, containing min and max IDs for some item types.
+		 * @param array $empty The modules with no items to sync during a full sync.
+		 *
+		 * @since 4.2.0
+		 * @since 7.3.0 Added $range arg.
+		 * @since 7.4.0 Added $empty arg.
+		 */
+		do_action( 'jetpack_full_sync_start', $full_sync_config, $range );
+		$this->send_action( 'jetpack_full_sync_start', array( $full_sync_config, $range ) );
+
+		return true;
+	}
+
+	/**
+	 * Whether full sync has started.
+	 *
+	 * @access public
+	 *
+	 * @return boolean
+	 */
+	public function is_started() {
+		return ! ! $this->get_status()['started'];
+	}
+
+	/**
+	 * Retrieve the status of the current full sync.
+	 *
+	 * @access public
+	 *
+	 * @return array Full sync status.
+	 */
+	public function get_status() {
+		$default = array(
+			'started'  => false,
+			'finished' => false,
+			'progress' => array(),
+			'config'   => array(),
+		);
+
+		return wp_parse_args( \Jetpack_Options::get_raw_option( self::STATUS_OPTION ), $default );
+	}
+
+	/**
+	 * Whether full sync has finished.
+	 *
+	 * @access public
+	 *
+	 * @return boolean
+	 */
+	public function is_finished() {
+		return ! ! $this->get_status()['finished'];
+	}
+
+	/**
+	 * Clear all the full sync data.
+	 *
+	 * @access public
+	 */
+	public function reset_data() {
+		$this->clear_status();
+		( new Lock() )->remove( self::LOCK_NAME );
+	}
+
+	/**
+	 * Clear all the full sync status options.
+	 *
+	 * @access public
+	 */
+	public function clear_status() {
+		\Jetpack_Options::delete_raw_option( self::STATUS_OPTION );
+	}
+
+	/**
+	 * Updates the status of the current full sync.
+	 *
+	 * @access public
+	 *
+	 * @param array $values New values to set.
+	 *
+	 * @return bool True if success.
+	 */
+	public function update_status( $values ) {
+		return $this->set_status( wp_parse_args( $values, $this->get_status() ) );
+	}
+
+	/**
+	 * Retrieve the status of the current full sync.
+	 *
+	 * @param array $values New values to set.
+	 *
+	 * @access public
+	 *
+	 * @return boolean Full sync status.
+	 */
+	public function set_status( $values ) {
+		return \Jetpack_Options::update_raw_option( self::STATUS_OPTION, $values );
+	}
+
+	/**
+	 * Given an initial Full Sync configuration set the initial status.
+	 *
+	 * @param array $full_sync_config Full sync configuration.
+	 *
+	 * @return array Initial Sent status.
+	 */
+	public function get_initial_progress( $full_sync_config ) {
+		// Set default configuration, calculate totals, and save configuration if totals > 0.
+		$status = array();
+		foreach ( $full_sync_config as $name => $config ) {
+			$module          = Modules::get_module( $name );
+			$status[ $name ] = array(
+				'total'    => $module->total( $config ),   // Total.
+				'sent'     => 0,             // Sent.
```

Comment:
```suggestion
				'sent'     => 0,
```
we use to not have keys for the array.